code/modules/projectiles/projectile.dm

```diff
@@ -686,6 +687,18 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 				BULLET_DEBUG("marksman_aura (+[shooter_human.marksman_aura * 3] + [proj.distance_travelled * shooter_human.marksman_aura * 0.35]).")
 				. += shooter_human.marksman_aura * 3
 				. += proj.distance_travelled * shooter_human.marksman_aura * 0.35
+	if(HAS_TRAIT(src, TRAIT_ISINTRENCH))
+		var/atom/D
+		for(D in proj.permutated)
+			if(istype(src, /mob/living/carbon/xenomorph) && !src.xenotrenchprotection)
+				break
+			BULLET_DEBUG("[proj.name] has passed through [D.loc.name]")
+			var/obj/structure/trench/S = locate() in D.loc
+			if(!S)
+				BULLET_DEBUG("mob trench acc penalty(65%).")
+				. *= 0.35 //35% chance to hit, which is a 65% protection
+				break
```

Comment:
This code makes no sense to me please explain

Reply:
Okay, this code makes it so that if the projectile at any point hits a mob in a trench, it will run a check in where to see if the projectile has ever left the trench. If it has, the trench modifier (35% to hit thing) will apply. If not, it hits like normal. **This way, you need line of sight in a trench to accurately shoot**

Reply:
I'll annotate it better so that it is less confusing